db/rate.go

```diff
@@ -0,0 +1,46 @@
+package db
+
+import (
+	"github.com/trustwallet/watchmarket/db/models"
+	"github.com/trustwallet/watchmarket/market"
+	"github.com/trustwallet/watchmarket/pkg/watchmarket"
+	"strconv"
+)
+
+func (i *Instance) AddRates(rates []watchmarket.Rate, provider market.Provider) error {
+
+	var errorsList []error
+
+	for _, rate := range rates {
+		r := models.Rate{
+			Rate:     rate,
+			Provider: provider.GetId(),
+		}
+
+		err := i.Gorm.Set("gorm:insert_option", "ON CONFLICT (id) DO NOTHING").Create(&r).Error
+		if err != nil {
+			errorsList = append(errorsList, err)
+		}
+	}
+
+	if len(errorsList) > 0 {
+		return ConvertToError(errorsList)
```

Comment:
I would suggest to return an array of errors here, this functions should not handle error managment

Reply:
👌🏻